csrc/mmdeploy/net/tvm/tvm_net.cpp

```diff
@@ -0,0 +1,288 @@
+        func_init(int(dev.device_type), int(dev.device_id), alloc_type);
+      }
+
+      // get input ids
+      auto func_input_index_ = vm_.GetFunction("get_input_index", false);
+      for (auto name : input_names) {
+        input_ids_[name] = func_input_index_(name, "main");
+      }
+
+      // get function
+      func_set_input_ = vm_.GetFunction("set_input");
+      func_run_ = vm_.GetFunction("invoke");
+    } else {
+      // graph executor won't do synchronize stream after run？
+      if (device_.is_device())
+        tvm::runtime::DeviceAPI::Get(dev)->SetStream(dev, stream_.GetNative());
+      tvm::runtime::Module gmod = mod_factory_.GetFunction("default")(dev);
+
+      // get function
+      func_set_input_ = gmod.GetFunction("set_input");
+      func_get_output_ = gmod.GetFunction("get_output");
+      func_run_ = gmod.GetFunction("run");
+    }
+
+    auto ToDesc = [&](const std::string& name) {
+      return TensorDesc{device_, DataType::kFLOAT, {}, name};
+    };
+
+    for (auto name : input_names) {
+      input_tensors_.emplace_back(ToDesc(name));
+    }
+
+    for (auto name : output_names) {
+      output_tensors_.emplace_back(ToDesc(name));
+    }
+
+  } catch (const std::exception& e) {
+    MMDEPLOY_ERROR("unhandled exception when creating TVM Net: {}", e.what());
+    return Status(eFail);
+  }
+
+  return success();
+}
+
+Result<void> TVMNet::ForwardAsync(Event* event) { return Status(eNotSupported); }
+
+Result<void> TVMNet::Deinit() { return success(); }
+
+Result<Span<Tensor>> TVMNet::GetInputTensors() { return input_tensors_; }
+
+Result<Span<Tensor>> TVMNet::GetOutputTensors() { return output_tensors_; }
+
+Result<void> TVMNet::Reshape(Span<TensorShape> input_shapes) {
+  for (size_t i = 0; i < input_shapes.size(); ++i) {
+    input_tensors_[i].Reshape(input_shapes[i]);
+  }
+  return success();
+}
+
+Result<void> TVMNet::Forward() {
+  DLDevice dev = GetDLDevice(device_);
+  try {
+    OUTCOME_TRY(stream_.Wait());
+
+    if (use_vm_) {
+      // vm
+
+      // set input
+      int num_inputs = input_tensors_.size();
+      std::vector<tvm::runtime::NDArray> args_arr(num_inputs);
+      std::vector<TVMValue> tvm_values(num_inputs + 1);
+      std::vector<int> tvm_type_codes(num_inputs + 1);
+      tvm::runtime::TVMArgsSetter setter(tvm_values.data(), tvm_type_codes.data());
+      setter(0, "main");
+      for (int k = 0; k < num_inputs; ++k) {
+        auto v = input_tensors_[k];
+        OUTCOME_TRY(auto managed_tensor, ToDLPack(v, stream_));
+        OUTCOME_TRY(stream_.Wait());
+        args_arr[k] = tvm::runtime::NDArray::FromDLPack(managed_tensor);
+
+        int input_id = input_ids_[v.name()];
+        setter(input_id + 1, args_arr[k]);
+      }
+      func_set_input_.CallPacked(
+          tvm::runtime::TVMArgs(tvm_values.data(), tvm_type_codes.data(), num_inputs + 1), nullptr);
+
+      // run
+      tvm::runtime::TVMRetValue ret = func_run_("main");
+      OUTCOME_TRY(stream_.Wait());
```

Comment:
The TVM vm seems to be running on the default stream. While `stream_` is not the CUDA default stream. The `Wait` has no effect here.

May use 
```C++
if (device_.is_device()) {
  OUTCOME_TRY(Stream(device_, nullptr).Wait());
}
```
to construct a wrapper on default stream and wait for it.

Reply:
what's the status now?